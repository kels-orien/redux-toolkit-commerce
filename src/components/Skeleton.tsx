import React from 'react'
import styled, { keyframes } from 'styled-components'




const Image = styled.div`
    height: 200px;
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
`

const shine = keyframes`
to {
    background-position-x: -200%;
  }
`;


const Card = styled.div`
  margin: 10px;
  width: 250px;
  border-radius: 5px;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.05);
  background: #eee;
  background: linear-gradient(110deg, #ececec 8%, #f5f5f5 18%, #ececec 33%);
  border-radius: 5px;
  background-size: 200% 100%;
  animation: 1.5s ${shine} linear infinite;
`

const Skeleton = () => {
    return (
        <Card>
            <Image />
        </Card>
    )
}

export default Skeleton
