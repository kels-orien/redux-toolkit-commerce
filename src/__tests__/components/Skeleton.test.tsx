import React from 'react'
import renderer from 'react-test-renderer'
import Skeleton from '../../components/Skeleton'

test('Skeleton render correctly', () => {
    const tree = renderer.create(<Skeleton/>).toJSON();

    expect(tree).toMatchSnapshot()

})