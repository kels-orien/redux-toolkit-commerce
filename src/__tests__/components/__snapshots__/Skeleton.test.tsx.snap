// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton render correctly 1`] = `
<div
  className="sc-gtsrHT jtGPnF"
>
  <div
    className="sc-bdnxRM gFJjPQ"
  />
</div>
`;
